tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,171 @@
+import copy
+import pytest
+import re
+
+from configuration import available_ports, TLS13_CIPHERS, ALL_TEST_CURVES, MINIMAL_TEST_CERTS
+from common import ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version, to_bytes
+
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+# arbitrarily large payload size
+PAYLOAD_SIZE = 1024
+
+# regex that matches server/client 'write to' with the TLS header
+OPENSSL_WRITE_TO_REGEX = r"write to [\w >\(\)\[\]=]*\\n0000 - 17 03 03 [0-9A-Fa-f]{2} [0-9A-Fa-f]{2}"
+
+
+def get_payload_size_from_openssl_trace(record: str) -> int:
+    # last five characters in trace is the record size in the form XX XX where X is a hex digit
+    size_in_hex = record[-5:].replace(' ', '')
+    size = int(size_in_hex, 16)
+    # record includes 16 bytes of aead tag
+    return size - 16
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+# only tls 1.3 supports record padding
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("padding_size", PADDING_SIZES, ids=get_parameter_name)
+def test_s2n_server_handles_padded_records(managed_process, cipher, provider, curve, protocol, certificate,
+                                           padding_size):
+    # Communication Timeline
+    # Client [OpenSSL]                       | Server [S2N]
+    # Handshake                              | Handshake
+    # Send padded data                       | Receive padded data
+    # Send padded close connection           | Close
```

Comment:
Nit: I'm not sure this comment is necessary / helpful. You're doing a very simple handshake + one round of app data. None of the framework's fun "markers" are explicitly involved.